code/modules/clothing/under/accessories/accessory.dm

```diff
@@ -384,6 +384,66 @@
 	icon_state = "stripedbluescarf"
 	item_color = "stripedbluescarf"
 
+//Necklaces
+/obj/item/clothing/accessory/necklace
+	name = "necklace"
+	desc = "A simple necklace."
+	icon_state = "necklace"
+	item_state = "necklace"
+	item_color = "necklace"
+	slot_flags = SLOT_MASK | SLOT_TIE
+
+/obj/item/clothing/accessory/necklace/locket
+	name = "gold locket"
+	desc = "A gold locket that seems to have space for a photo within."
+	icon_state = "locket"
+	item_state = "locket"
+	item_color = "locket"
+	slot_flags = SLOT_MASK | SLOT_TIE
+	var/base_icon
+	var/open
+	var/obj/item/held //Item inside locket.
+
+/obj/item/clothing/accessory/necklace/locket/Destroy()
+	QDEL_NULL(held)
+	return ..()
+
+
+/obj/item/clothing/accessory/necklace/locket/attack_self(mob/user as mob)
+	if(!base_icon)
+		base_icon = icon_state
+
+	if(!("[base_icon]_open" in icon_states(icon)))
+		to_chat(user, "The [src] doesn't seem to open.")
+		return
+
+	open = !open
+	to_chat(user, "You flip the [src] [open?"open":"closed"].")
+	if(open)
+		icon_state = "[base_icon]_open"
+		if(held)
+			to_chat(user, "The [held] falls out!")
+			held.forceMove(get_turf(user))
+			held = null
+	else
+		icon_state = "[base_icon]"
+
+/obj/item/clothing/accessory/necklace/locket/attackby(var/obj/item/O as obj, mob/user as mob)
+	if(!open)
+		to_chat(user, "You have to open it first.")
+		return
+
+	if(istype(O,/obj/item/weapon/paper) || istype(O, /obj/item/weapon/photo) && !(istype(O, /obj/item/weapon/paper/talisman)))
+		if(held)
+			to_chat(usr, "The [src] already has something inside it.")
```

Comment:
`[src]` will automatically add "the" as needed... or is supposed to. Never truly expect byond to work as advertised since they don't advertise the code to work even half the time.